sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulseConfiguration.java

```diff
@@ -0,0 +1,129 @@
+package com.azure.monitor.opentelemetry.exporter.implementation.quickpulse;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.util.logging.ClientLogger;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+
+import java.util.ArrayList;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+
+
+public class QuickPulseConfiguration {
+    private static final ClientLogger logger = new ClientLogger(QuickPulseDataFetcher.class);
+    private  AtomicReference<String> etag = new AtomicReference<>();
+    private ConcurrentHashMap<String, ArrayList<OpenTelMetricInfo>> metrics = new ConcurrentHashMap<>();
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    public synchronized String getEtag() {
+        return this.etag.get();
+    }
+
+    public synchronized void setEtag(String etag) {
+        this.etag.set(etag);
+    }
+
+    public synchronized ConcurrentHashMap<String, ArrayList<OpenTelMetricInfo>> getMetrics() {
+        return this.metrics;
+    }
+
+    public synchronized void setMetrics(ConcurrentHashMap<String, ArrayList<OpenTelMetricInfo>> metrics) {
+        this.metrics = metrics;
+    }
+
+    public synchronized void updateConfig(String etagValue, ConcurrentHashMap<String, ArrayList<OpenTelMetricInfo>> otelMetrics) {
+        if (!Objects.equals(this.getEtag(), etagValue)){
+            this.setEtag(etagValue);
+            this.setMetrics(otelMetrics);
+        }
+
+    }
+
+    public ConcurrentHashMap<String,  ArrayList<OpenTelMetricInfo>> parseMetrics(HttpResponse response) {
```

Comment:
Also related to my other comments about response containing derived metric infos of different telemetry types - suggestion on how to change this method to not mix up telemetry types:

- inside the forloop, I wouldn't initialize OpenTelMetricInfo in the beginning. 
- Only do the logic to create the OpenTelMetricInfo, add to requested metrics, if the TelemetryType == "Metric"
- I don't see a need to return the requested metrics map here; you can save it directly once its computed as all the computation is happening inside the same class.

Reply:
for flexibility down the line when live filtering is enabled, I decided to allow parsing of any derived metric object. The SDK still only responds to derived metrics of "Metric" telemetry type accordingly.